test/new-e2e/tests/agent-shared-components/api/api_test.go

```diff
@@ -106,10 +314,102 @@ func (v *apiSuite) TestInternalAgentAPIEndpoints() {
 	for _, tc := range testcases {
 		cmd := tc.fetchCommand(authtoken)
 		host := v.Env().RemoteHost
-		require.EventuallyWithT(v.T(), func(t *assert.CollectT) {
-			resp, err := host.Execute(cmd)
-			require.NoError(t, err)
-			assert.Contains(t, resp, tc.want, "%s %s returned: %s, wanted: %s", tc.method, tc.endpoint, resp, tc.want)
-		}, 2*time.Minute, 10*time.Second)
+		v.T().Run(fmt.Sprintf("API - %s test", tc.name), func(t *testing.T) {
+			require.EventuallyWithT(t, func(ct *assert.CollectT) {
+				resp, err := host.Execute(cmd)
+				require.NoError(ct, err)
```

Comment:
I believe `require` within `EventuallyWithT` does not behave as we would expect. If `require` fails, the whole `EventuallyWithT` will fail even if we have time for more ticks.

I think we should use something like `if assert.NoErrror(ct, err) {`

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/25000/commits/4496c36d67d904500f3638a31c5c24fa0921d50a